.github/workflows/ci.yml

```diff
@@ -32,6 +32,22 @@ jobs:
         with:
           keyword: "[skip-ci]"
 
+  pre-commit:
+    name: Format
+    runs-on: ubuntu-latest
+    needs: detect-ci-trigger
+    if: needs.detect-ci-trigger.outputs.triggered == 'false'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+      - uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: --hook-stage manual --all-files
+
```

Comment:
we should probably avoid using `pre-commit/action` which is in maintenance-only mode and replaced by either `pre-commit.ci` or `pre-commit-ci/lite-action` (I usually use the former because it periodically opens PRs whenever there's new versions of hooks, where the available frequencies are "weekly", "monthly", "quarterly")

Reply:
sounds good, feel free to push to this PR!

Reply:
then I'd probably go with `pre-commit.ci` because that has more features (autoupdate is really a convenient feature). @mdhaber, for that to work you'll have to set up https://github.com/marketplace/pre-commit-ci, and as this is a personal repository and I'm not a repository admin I can't do it for you.

Let's do all the changes in a separate PR though (see https://github.com/mdhaber/marray/pull/46#issuecomment-2564092685)

Reply:
I've reverted all the unrelated changes, let's continue this in #48